src/openloco/windows/stationwnd.cpp

```diff
@@ -1,55 +1,660 @@
+            buffer = stringmgr::format_string(buffer, string_ids::accepted_cargo_separator);
+
+            auto station = stationmgr::get(self->number);
+            uint8_t cargoTypeCount = 0;
+
+            for (uint32_t cargoId = 0; cargoId < max_cargo_stats; cargoId++)
+            {
+                auto& stats = station->cargo_stats[cargoId];
+
+                if (!stats.is_accepted())
+                    continue;
+
+                *buffer++ = ' ';
+                *buffer++ = control_codes::inline_sprite_str;
+                *(reinterpret_cast<uint32_t*>(buffer)) = objectmgr::get<cargo_object>(cargoId)->unit_inline_sprite;
+                buffer += 4;
+
+                cargoTypeCount++;
+            }
+
+            if (cargoTypeCount == 0)
+            {
+                buffer = stringmgr::format_string(buffer, string_ids::cargo_nothing_accepted);
+            }
+
+            *buffer++ = '\0';
+            auto args = FormatArguments();
+            args.push(string_ids::buffer_1250);
+            
+            const auto& widget = self->widgets[widx::status_bar];
+            const auto x = self->x + widget.left - 1;
+            const auto y = self->y + widget.top - 1;
+            const auto width = widget.width();
+            gfx::draw_string_494BBF(*dpi, x, y, width, colour::black, string_ids::buffer_1250, &args);
+        }
+
+        // 0x0048EB0B
+        static void on_mouse_up(window* self, widget_index widgetIndex)
+        {
+            switch (widgetIndex)
+            {
+                case common::widx::caption:
+                    common::renameStationPrompt(self, widgetIndex);
+                    break;
+
+                case common::widx::close_button:
+                    WindowManager::close(self);
+                    break;
+
+                case common::widx::tab_station:
+                case common::widx::tab_cargo:
+                case common::widx::tab_cargo_ratings:
+                    common::switchTab(self, widgetIndex);
+                    break;
+
+                case widx::station_catchment:
+                    if (self->number == word_112C786)
+                        self->number = 65535;
+
+                    // 0x0049271A
+                    registers regs;
+                    regs.ax = self->number;
+                    call(0x0049271A, regs);
+                    break;
+            }
+        }
+
+        // 0x0048EBB7
+        static void on_resize(window* self)
+        {
+            // Call to sub_48E32C has been deliberately omitted.
+
+            self->set_size(gfx::ui_size_t(299, 172), gfx::ui_size_t(299, 327));
+        }
+
+        // 0x0048EB64
+        static void get_scroll_size(window* self, uint32_t scrollIndex, uint16_t* scrollWidth, uint16_t* scrollHeight)
+        {
+            auto station = stationmgr::get(self->number);
+            *scrollHeight = 0;
+            for (uint8_t i = 0; i < 32; i++)
```

Comment:
I think you could use a ranged for here. `for (const auto& cargoStats : station->cargo_stats)`